typescript/sdk/src/registry/GithubRegistry.ts

```diff
@@ -0,0 +1,145 @@
+import { Logger } from 'pino';
+import { parse } from 'yaml';
+
+import { rootLogger } from '@hyperlane-xyz/utils';
+
+import { ChainMetadata } from '../metadata/chainMetadataTypes.js';
+import { ChainMap, ChainName } from '../types.js';
+
+import { ChainAddresses, IRegistry } from './IRegistry.js';
+
+const DEFAULT_REGISTRY = 'https://github.com/hyperlane-xyz/hyperlane-registry';
+const CHAIN_FILE_REGEX = /chains\/([a-z]+)\/([a-z]+)\.yaml/;
+
+export interface GithubRegistryOptions {
+  url?: string;
+  branch?: string;
+  authToken?: string;
+  logger?: Logger;
+}
+
+interface TreeNode {
+  path: string;
+  type: 'blob' | 'tree';
+  size?: number;
+  sha: string;
+  url: string;
+}
+
+interface ChainFiles {
+  metadata?: true;
+  addresses?: true;
+}
+
+interface RepoContent {
+  chains: ChainMap<ChainFiles>;
+  // TODO add deployment artifacts here
+}
```

Comment:
this feels a bit like a recursive dependency with registry internals but idk if theres an alternative